src/gg.regression.unity.bots/Runtime/Scripts/RGOverlay/RGSequenceEditor.cs

```diff
@@ -168,13 +206,24 @@ public int ClearAvailableSegments()
 
         /**
          * <summary>
-         * Save the current Sequence to disk
+         * Save or update the current Sequence
          * </summary>
          */
         public void SaveSequence()
         {
-            var addedSegments = _dropZone.GetChildren();
+            if (!string.IsNullOrEmpty(_existingSequencePath))
+            {
+                CurrentSequence.segments.Clear();
+
+                // If the Sequence being edited is renamed, we must delete the original Sequence
+                // first. Else we will end up with a duplicate
+                if (CurrentSequence.name != NameInput.text)
+                {
+                    BotSequence.DeleteSequenceAtPath(_existingSequencePath);
+                }
```

Comment:
A bit more info (beyond my Loom) on why the existing Sequence gets deleted before saving a new one, if the name changed. We generate the Sequence's file name based on the Sequence's name, so a changed name would result in a different file name. This is why I delete the original